modules/file/file.go

```diff
@@ -835,16 +835,22 @@ func (r *run) pathWalk(path string, roots []string) (traversed []string, err err
 			entryAbsPath := path
 			// append path separator if missing
 			if entryAbsPath[len(entryAbsPath)-1] != os.PathSeparator {
-				entryAbsPath += string(os.PathSeparator)
+				// if not a symlink
+				if dirEntry.Mode()&os.ModeSymlink != os.ModeSymlink {
+					entryAbsPath += string(os.PathSeparator)
+				}
 			}
 			entryAbsPath += dirEntry.Name()
 			// this entry is a subdirectory, keep it for later
 			if dirEntry.IsDir() {
-				// append trailing slash
-				if entryAbsPath[len(entryAbsPath)-1] != os.PathSeparator {
-					entryAbsPath += string(os.PathSeparator)
+				// don't follow directory symlink
+				if dirEntry.Mode()&os.ModeSymlink != os.ModeSymlink {
```

Comment:
Yes, this allow to not put symlinked sub directories in subdir and fix real dir path. But I think that the other bloc in 838 is useless.